src/Avalonia.Interactivity/InteractiveExtensions.cs

```diff
@@ -57,5 +57,55 @@ public static class InteractiveExtensions
                 routes,
                 handledEventsToo));
         }
+
+        /// <summary>
+        /// Builds an event route for a routed event.
+        /// </summary>
+        /// <param name="interactive">The interactive element to start building the route at.</param>
+        /// <param name="e">The routed event.</param>
+        /// <returns>An <see cref="EventRoute"/> describing the route.</returns>
+        /// <remarks>
+        /// Usually, calling <see cref="RaiseEvent(IInteractive, RoutedEventArgs)"/> is sufficent to raise a routed
+        /// event, however there are situations in which the construction of the event args is expensive
+        /// and should be avoided if there are no handlers for an event. In these cases you can call
+        /// this method to build the event route and check the <see cref="EventRoute.HasHandlers"/>
+        /// property to see if there are any handlers registered on the route. If there are, call
+        /// <see cref="EventRoute.RaiseEvent(IInteractive, RoutedEventArgs)"/> to raise the event.
+        /// </remarks>
+        public static EventRoute BuildEventRoute(this IInteractive interactive, RoutedEvent e)
```

Comment:
Just a question. Why this method was made as an extension method? It can be also moved to EventHandlers.cs to avoid breaking change in IInteractive event.
Or the reason was to not add this method to EventHandres.cs, because it is a bit higher abstraction that works with interactive-tree only?

Reply:
Yeah, it also would still be breaking since EventHandlers isn't part of the inheritance chain :-|